moped-editor/src/views/projects/projectView/ProjectComponents/mapStyleSettings.js

```diff
@@ -18,240 +18,278 @@ export const COLORS = {
-      "circle-stroke-opacity": 0.9,
-      "circle-color": COLORS.pinkLight,
-      "circle-opacity": 0.9,
     },
   },
   "project-lines": {
-    id: "project-lines",
-    _featureIdProp: "CTN_SEGMENT_ID",
-    type: "line",
-    paint: {
-      "line-width": {
-        stops: [
-          [10, 2],
-          [20, 10],
+    isInteractive: true,
+    layerProps: {
+      id: "project-lines",
+      _featureIdProp: "CTN_SEGMENT_ID",
+      type: "line",
+      paint: {
+        "line-width": {
+          stops: [
+            [10, 2],
+            [20, 10],
+          ],
+        },
+        "line-color": [
+          "case",
+          ["boolean", ["feature-state", "hover"], false],
+          COLORS.pinkDark,
+          COLORS.pinkBright,
         ],
       },
-      "line-color": [
-        "case",
-        ["boolean", ["feature-state", "hover"], false],
-        COLORS.pinkDark,
-        COLORS.pinkBright,
-      ],
-    },
-    layout: {
-      "line-cap": "round",
+      layout: {
+        "line-cap": "round",
+      },
     },
   },
   "project-points-muted": {
-    id: "project-points",
-    _featureIdProp: "INTERSECTION_ID",
-    type: "circle",
-    paint: {
-      "circle-radius": {
-        stops: [
-          [10, 1],
-          [20, 12],
-        ],
+    isInteractive: false,
+    layerProps: {
+      id: "project-points",
+      _featureIdProp: "INTERSECTION_ID",
+      type: "circle",
+      paint: {
+        "circle-radius": {
+          stops: [
+            [10, 1],
+            [20, 12],
+          ],
+        },
+        "circle-stroke-color": COLORS.mutedGray,
+        "circle-stroke-width": 2,
+        "circle-stroke-opacity": 0.9,
+        "circle-color": COLORS.mutedGray,
+        "circle-opacity": 0.9,
       },
-      "circle-stroke-color": COLORS.mutedGray,
-      "circle-stroke-width": 2,
-      "circle-stroke-opacity": 0.9,
-      "circle-color": COLORS.mutedGray,
-      "circle-opacity": 0.9,
     },
   },
   "project-lines-muted": {
-    id: "project-lines",
-    _featureIdProp: "CTN_SEGMENT_ID",
-    type: "line",
-    paint: {
-      "line-width": {
-        stops: [
-          [10, 2],
-          [20, 10],
-        ],
+    isInteractive: false,
+    layerProps: {
+      id: "project-lines",
+      _featureIdProp: "CTN_SEGMENT_ID",
+      type: "line",
+      paint: {
+        "line-width": {
+          stops: [
+            [10, 2],
+            [20, 10],
+          ],
+        },
+        "line-color": COLORS.mutedGray,
+      },
+      layout: {
+        "line-cap": "round",
       },
-      "line-color": COLORS.mutedGray,
-    },
-    layout: {
-      "line-cap": "round",
     },
   },
   "project-lines-underlay": {
-    id: "project-lines-underlay",
-    _featureIdProp: "CTN_SEGMENT_ID",
-    type: "line",
-    paint: {
-      "line-width": 20,
-      "line-color": COLORS.white,
-      "line-opacity": 0,
-    },
-    layout: {
-      "line-cap": "round",
+    isInteractive: false,
```

Comment:
@mddilley in the previous version, we were using the project and ctn "underlays" as the interactive layers. these invisible layers just help with the mouse interaction by making the clickable area of each line wider. it feels like a bit of a hack, but i definitely prefer the click interaction when we use them.

Reply:
That makes perfect sense about why I felt like I made it harder to select segments somehow. 🙃 I'll gladly add that back in. 🙏